types/rdf-dataset-indexed/dataset.d.ts

```diff
@@ -0,0 +1,34 @@
+import { DatasetCore, DatasetFactory, DataFactory, Quad, BaseQuad, Stream, Term } from 'rdf-js';
+import { Readable } from 'stream';
+
+declare namespace DatasetIndexed {
+    interface DatasetIndexed<Q extends BaseQuad = Quad, InQuad extends BaseQuad = Quad> extends DatasetCore<Q, InQuad> {
+        create(quads: DatasetIndexed<BaseQuad, BaseQuad> | BaseQuad[]): this;
```

Comment:
This (and others) probably take a `DatasetCore<InQuad ,Q>` rather than just a `DatasetIndexed`?

Reply:
Unfortunately not. Underneath the `N3Store` calls `_addQuads` which either iterates an array or calls `toArray`, the latter not being member of `DatasetCore`

https://github.com/rdf-ext/rdf-dataset-indexed/blob/7ea6c12626be211080c25af6c6eb853318474fbc/lib/N3Store.js#L300-L304

Reply:
Ah, hadn't spotted that.